app/services/shelf_list.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+# @abstract Returns a hash with two sets of ShelfItem objects for a given call number query. The sets are divided
+# between those items that come after a given query parameter, and those that come before. The order of each is the raw
+# order coming out of Solr.  Any additional processing or arrangement is left to whichever class or method is calling
+# this service.
+
+class ShelfList
+  # @return [Hash]
+  # @note Returns a hash of after and before list items.
+  def self.call(args)
+    new(args).documents
+  end
+
+  attr_reader :query, :forward_limit, :reverse_limit, :forward_shelfkey, :reverse_shelfkey
+
+  def initialize(
+    query:,
+    forward_limit:,
+    reverse_limit:,
+    forward_shelfkey: 'forward_lc_shelfkey',
+    reverse_shelfkey: 'reverse_lc_shelfkey'
+  )
+    @query = query
+    @reverse_limit = (reverse_limit || 10).to_i
+    @forward_limit = (forward_limit || 10).to_i
+    @forward_shelfkey = forward_shelfkey
+    @reverse_shelfkey = reverse_shelfkey
+  end
+
+  def documents
+    {
+      after: holdings(keys: forward_keys, shelfkey: forward_shelfkey, direction: 'forward'),
+      before: holdings(keys: reverse_keys, shelfkey: reverse_shelfkey, direction: 'reverse')
+    }
+  end
+
+  private
+
+    def forward_keys
+      @forward_keys ||= terms_query(field: forward_shelfkey, limit: forward_limit)
+        .each_slice(2)
+        .map(&:first)
+    end
+
+    def reverse_keys
+      @reverse_keys ||= terms_query(
+        field: reverse_shelfkey,
+        value: ShelfKey.reverse(query),
+        limit: reverse_limit
+      )
+        .each_slice(2)
+        .map(&:first)
+    end
+
+    # @return Array<ShelfItem>
+    # @note Uses a Holdings object to build a set of shelf item objects from #document_query. A set of documents is
+    # retrieved from Solr for a given set of shelf keys. The keys are already in the proper order, we just need a
+    # complete document for each one. Using Holdings, we construct a hash of every shelf key and its document from
+    # #document_query.  There can be multiple documents per key, if the same call number is present in multiple items.
+    def holdings(keys:, shelfkey:, direction:)
```

Comment:
@awead I kinda wanna rename this method since there is the Holdings object, but I do not have a better idea than holdings :)

Reply:
`shelf_items`?

Reply:
yeah that is exactly what I thought as an option too. So yeah I kinda like that, if you are OK with it too

Reply:
@banukutlu all fixed up. This was a good catch. I was already using `holdings` as a variable internal to the method, which is bad style.

Reply:
awesome, thanks 😎